doc/api/process.md

```diff
@@ -1684,6 +1684,107 @@ the [`'warning'` event][process_warning] and the
 [`emitWarning()` method][process_emit_warning] for more information about this
 flag's behavior.
 
+## `process.parseArgs([argv[, options]])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `argv` {Array<string>|Object} (Optional) Array of argument strings; defaults
+  to [`process.argv.slice(2)`](process_argv). If an Object, the default is used,
+  and this parameter is considered to be the `options` parameter.
+* `options` {Object} (Optional) The `options` parameter, if present, is an
+  object supporting the following property:
+  * `expectsValue` {Array<string>|string} (Optional) One or more argument
+    strings which _expect a value_ when present
+* Returns: {Object} An object having properties corresponding to parsed Options
+  and Flags, and a property `_` containing Positionals
+
+The `process.parseArgs` function parses command-line arguments from an array of
+strings and returns an object representation.
+
+Example using [`process.argv`][]:
+
+```js
+// script.js
+// called via `node script.js --foo bar baz`
+const argv = process.parseArgs();
+
+// argv.foo === true
+if (argv.foo) {
+  console.log(argv._); // prints [ 'bar', 'baz' ]
+}
+```
+
+Example using a custom `argv` and the `expectsValue` option:
+
+```js
+const argv = process.parseArgs(
+  ['--foo', 'bar', 'baz'],
+  { expectsValue: ['foo'] }
+);
+
+// argv.foo === 'bar'
+if (argv.foo === 'bar') {
+  console.log(argv._); // prints [ 'baz' ]
+}
+```
+
+[`ERR_INVALID_ARG_TYPE`][] will be thrown if the `argv` parameter is not an
+Array.
+
+Arguments fall into one of three catgories:
+
+* _Flags_, which begin with one or more dashes (`-`), and _do not_ have an
+  associated string value (e.g., `node app.js --verbose`)
+  * These will be parsed automatically; you do not need to "declare" them
+  * The Flag _name_ is the string following the prefix of one-or-more dashes,
+    e.g., the name of `--foo` is `foo`
+  * Flag names become property names in the returned object
+  * When appearing _once_ in the array, the value of the property will be `true`
+  * When _repeated_ in the array, the value of the property becomes a count of
+    repetitions (e.g., `['-v', '-v' '-v']` results in `{ v: 3 }`)
```

Comment:
This has to go IMO 😅

Reply:
I'm not sure we should remove it. This would completely prevent users needing the count from using our API.

Reply:
Is that a thing CLIs provide? I have never seen any CLI behaving that way and I can't think of any use cases where it would be relevant

Reply:
I have seen this pattern some times to increase verbosity in tools like:
- [ssh](https://linux.die.net/man/1/ssh)
- [ansible](https://docs.ansible.com/ansible/latest/cli/ansible.html#cmdoption-ansible-v)

I don't know if it's really common, but my point is that if we don't support it somehow, there would be no way for the user of our API to support it. It's different from other use cases like throwing on unknown options, or converting value types. Those can be built upon the proposed API.

Reply:
Right, but we're also leaving out the single dash with multiple flags feature which is way more popular than this one. I still think it's a weird behavior but won't block the PR because of it.

Reply:
One feature I don't think this proposal supports, which I know is quite popular, is the concept of arrays:

```
my-script -f file-1.js -f file2.js
```

or,

```
my-script --file file-1.js --file file-2.js
```

I  wonder if instead of introducing  this special handling of `-v`, we could instead add an additional configuration  setting `array`, which would  indicate the above behavior. Someone could then do this:

```
my-script -v -v -v
``` 

which would output `argv.v = [true, true, true]`.

☝️ you can just take the length of this array.

Reply:
`-v -v -v` convinced me that we should either keep the counter or return an array of truths

Reply:
I mean, I could do this:

```js
util.parseArgs(['-v']) // {v: 1}
util.parseArgs(['-v', '-v', '-v]) // {v: 3}
```

then it's always a number.  feels _really_ weird to me, but at least it's consistent.

FWIW, `Number(true)` is `1`, so not a huge deal if you must have a number.  I think people will expect `true` 98% of the time.